src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourceMappingOptionsControl.xaml.cs

```diff
@@ -0,0 +1,250 @@
+        }
+
+        private bool CanExecuteClearButtonCommand(object parameter)
+        {
+            return SourceMappingsCollection.Count > 0;
+        }
+
+        internal bool ApplyChangedSettings()
+        {
+            IReadOnlyDictionary<string, IReadOnlyList<string>> Patterns = new Dictionary<string, IReadOnlyList<string>>();
+            PackageSourceMapping packageSourceMappings = new PackageSourceMapping(Patterns);
+            var componentModel = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            var settings = componentModel.GetService<ISettings>();
+            PackageSourceMappingProvider packageSourceMappingProvider = new PackageSourceMappingProvider(settings);
+            IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingsSourceItems = ReadMappingsFromUIToConfig(SourceMappingsCollection);
+            try
+            {
+                if (SourceMappingsChanged(_originalPackageSourceMappings, packageSourceMappingsSourceItems))
+                {
+                    packageSourceMappingProvider.SavePackageSourceMappings(packageSourceMappingsSourceItems);
+                }
+            }
+            // Thrown during creating or saving NuGet.Config.
+            catch (NuGetConfigurationException ex)
+            {
+                MessageHelper.ShowErrorMessage(ex.Message, Resx.ErrorDialogBoxTitle);
+                return false;
+            }
+            // Thrown if no nuget.config found.
+            catch (InvalidOperationException ex)
+            {
+                MessageHelper.ShowErrorMessage(ex.Message, Resx.ErrorDialogBoxTitle);
+                return false;
+            }
+            catch (UnauthorizedAccessException)
+            {
+                MessageHelper.ShowErrorMessage(Resx.ShowError_ConfigUnauthorizedAccess, Resx.ErrorDialogBoxTitle);
+                return false;
+            }
+            // Unknown exception.
+            catch (Exception ex)
+            {
+                MessageHelper.ShowErrorMessage(Resx.ShowError_ApplySettingFailed, Resx.ErrorDialogBoxTitle);
+                ActivityLog.LogError(NuGetUI.LogEntrySource, ex.ToString());
+                return false;
+            }
+            return true;
+        }
+
+        // Returns true if there are changes between existingSourceMappings and packageSourceMappings.
+        private static bool SourceMappingsChanged(IReadOnlyList<PackageSourceMappingSourceItem> existingSourceMappings, IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappings)
+        {
+            if (existingSourceMappings.Count != packageSourceMappings.Count)
+            {
+                return true;
+            }
+
+            for (int i = 0; i < existingSourceMappings.Count; ++i)
+            {
+                //checks to see if keys match
+                if (existingSourceMappings[i] != packageSourceMappings[i])
```

Comment:
Currently it's not comparing the keys, it's just comparing 2 objects in memory.

```suggestion
                if (existingSourceMappings[i].Key != packageSourceMappings[i].Key)
```

Reply:
I don't think I need to add the ".Key" because both of these are PackageSourceMappingSourceItems which have and Equals() method that compares the keys